src/resources/ble_darwin.mm

```diff
@@ -0,0 +1,1297 @@
+  bool active = [central_manager->central_manager() isScanning];
+
+  if (active || central_manager->scan_active()) ALREADY_IN_USE;
+  central_manager->set_scan_active(true);
+  [central_manager->central_manager() scanForPeripheralsWithServices:nil options:nil];
+
+  run_async([=]() -> void {
+    LightLocker locker(central_manager->scan_mutex());
+    OS::wait_us(central_manager->stop_scan_condition(), duration_us);
+    [central_manager->central_manager() stopScan];
+    central_manager->set_scan_active(false);
+    HostBLEEventSource::instance()->on_event(central_manager, kBLECompleted);
+  });
+
+  return process->program()->null_object();
+}
+
+PRIMITIVE(scan_next) {
+  ARGS(BLECentralManagerResource, central_manager);
+
+  DiscoveredPeripheral* peripheral = central_manager->next_discovered_peripheral();
+  if (!peripheral) return process->program()->null_object();
+
+  Array* array = process->object_heap()->allocate_array(6, process->program()->null_object());
+  if (!array) ALLOCATION_FAILED;
+
+  const char* address = [[[peripheral->peripheral() identifier] UUIDString] UTF8String];
+  String* address_str = process->allocate_string(address);
+  if (address_str == null) {
+    delete peripheral;
+    ALLOCATION_FAILED;
+  }
+  array->at_put(0, address_str);
+
+  array->at_put(1, Smi::from([peripheral->rssi() shortValue]));
+
+  NSString* identifier = [peripheral->peripheral() name];
+  if (identifier != nil) {
+    String* identifier_str = process->allocate_string([identifier UTF8String]);
+    if (identifier_str == null) {
+      free(peripheral);
+      ALLOCATION_FAILED;
+    }
+    array->at_put(2, identifier_str);
+  }
+
+  NSArray* discovered_services = peripheral->services();
+  if (discovered_services != nil) {
+    Array* service_classes = process->object_heap()->allocate_array(
+        static_cast<int>([discovered_services count]),
+        process->program()->null_object());
+
+    for (int i = 0; i < [discovered_services count]; i++) {
+      String* uuid = process->allocate_string([[discovered_services[i] UUIDString] UTF8String]);
+      if (uuid == null) {
+        free(peripheral);
+        ALLOCATION_FAILED;
+      }
+      service_classes->at_put(i, uuid);
+    }
+    array->at_put(3, service_classes);
+  }
+
+  NSData* manufacturer_data = peripheral->manufacturer_data();
+  if (manufacturer_data != nil) {
+    ByteArray* custom_data = process->object_heap()->allocate_internal_byte_array(
+        static_cast<int>([manufacturer_data length]));
+    ByteArray::Bytes custom_data_bytes(custom_data);
+    if (!custom_data) ALLOCATION_FAILED;
+    memcpy(custom_data_bytes.address(), manufacturer_data.bytes, [manufacturer_data length]);
+
+    array->at_put(4, custom_data);
+  }
+
+  NSNumber* is_connectable = peripheral->connectable();
+  array->at_put(5, (is_connectable != nil && is_connectable.boolValue == YES)
+                   ? process->program()->true_object()
+                   : process->program()->false_object());
```

Comment:
I believe this works.
```suggestion
  array->at_put(5, BOOL(is_connectable != nil && is_connectable.boolValue == YES));
```

Reply:
You should try it :)
The issue is that exactly for objective-c, they have their own BOOL macro....

Reply:
Aah. you undefed the Toit one in line 24, and not the macos one...